ikarus/python/finiteelements/material.hh

```diff
@@ -17,131 +17,187 @@
-    using beamMaterialClass = decltype(beamMaterial(std::declval<Materialname>()));                                            \
-    auto beammaterial       = Dune::Python::insertClass<beamMaterialClass>(                                                    \
-                            scope, std::string("Beam_") + #Materialname,                                                 \
-                            Dune::Python::GenerateTypeName(                                                              \
-                                "Ikarus::VanishingStress<std::array<Ikarus::Impl::StressIndexPair, "                     \
-                                      "2ul>{{Impl::StressIndexPair{1, 1},Ikarus::Impl::StressIndexPair{2ul, 2ul}}}," +         \
-                                Dune::className<Materialname>() + ">"),                                                  \
-                            includes)                                                                                    \
-                            .first;                                                                                            \
-    MAKE_MaterialFunction(beammaterial, beamMaterialClass, storedEnergy, 4);                                                   \
-    MAKE_MaterialFunction(beammaterial, beamMaterialClass, stresses, 4);                                                       \
-    MAKE_MaterialFunction(beammaterial, beamMaterialClass, tangentModuli, 4);                                                  \
-    MAKE_MaterialFunction(beammaterial, beamMaterialClass, storedEnergy, 6);                                                   \
-    MAKE_MaterialFunction(beammaterial, beamMaterialClass, stresses, 6);                                                       \
-    MAKE_MaterialFunction(beammaterial, beamMaterialClass, tangentModuli, 6);                                                  \
-    cls##Materialname.def("asBeamMaterial", [](Materialname& self) {                                                           \
-      return beamMaterial(self);                                                                                               \
-    }); /* no keep_alive since beamMaterial copies the material */                                                             \
+namespace Ikarus::Python {
+
+namespace Impl {
+  // Function to extract and convert parameters using a conversion strategy
+  template <typename T>
+  LamesFirstParameterAndShearModulus convertMaterialParameters(const pybind11::kwargs& kwargs,
+                                                               const std::string& param1, const std::string& param2) {
+    auto converter =
+        convertLameConstants(T{kwargs[param1.c_str()].cast<double>(), kwargs[param2.c_str()].cast<double>()});
+
+    // This is necessary as the converter cannnot convert to a parameter already present due to compile-time constraints
```

Comment:
Didi this typo reeappear? I'm on the most recent version I suppose.

Reply:
Wierd in my current branch its fixed